torch/distributed/device_mesh.py

```diff
@@ -183,44 +182,51 @@ class DeviceMesh:
         def __init__(
             self,
             device_type: str,
-            mesh: Union[torch.Tensor, "ArrayLike"],
+            mesh: Optional[Union[torch.Tensor, "ArrayLike"]] = None,
```

Comment:
Can we update the docstring to explain when `mesh` can be None? Even if we just preserve it for the internal usage, we should mention because this is a public API.